### ISSUE TITLE
Relax restrictions when `_id` is not the first field in projection

### ISSUE THREAD

**Issue by b1ron** [FerretDB/FerretDB#2826]
### FerretDB version

v1.3.0

Currently we only relax projection restrictions when `_id` is the first field in the projection spec. We should allow it be at any index, just like MongoDB.

### Backend

Any.

### Environment

- OS: Ubuntu 22.10 x86_64
- Deployment: v1.3.0
- Deployment details: 23.0.3

### What did you do?

```sh
mongosh --quiet --eval 'db.foo.insertOne({a: 1}); db.foo.find({}, {a: 1, _id: 0});'
```

### What did you expect to see?

```
[ { a: 1 } ]
```

### What did you see instead?

```
MongoServerError: Cannot do exclusion on field _id in inclusion projection
```

**Comment by DkGr** [comment]
These restrictions make unusable the MongoDB.Entities library for .NET. All simple operations which need to get some data from a collection (ex: Find, Update, Where or Download) always return "MongoServerError: Cannot do exclusion on field _id in inclusion projection" because the code-behind from the MongoDB.Entities library uses explicit projection with the official MongoDB C# driver.

**Comment by BoBiene** [comment]
I have the same issue with the MongoDB C# Linq driver when using the .Select(... ) with the current FerritDB 1.8 debug docker (postgres)

````
2023-08-16T21:30:28.117Z    DEBUG   // 172.17.0.1:39752 -> 172.17.0.2:27017         clientconn/conn.go:288  Request message:
2023-08-16 23:30:28 {
2023-08-16 23:30:28   "Checksum": 0,
2023-08-16 23:30:28   "FlagBits": 0,
2023-08-16 23:30:28   "Sections": [
2023-08-16 23:30:28     {
2023-08-16 23:30:28       "Document": {
2023-08-16 23:30:28         "$k": [
2023-08-16 23:30:28           "aggregate",
2023-08-16 23:30:28           "pipeline",
2023-08-16 23:30:28           "cursor",
2023-08-16 23:30:28           "$db"
2023-08-16 23:30:28         ],
2023-08-16 23:30:28         "aggregate": "ConnectionGroup",
2023-08-16 23:30:28         "pipeline": [
2023-08-16 23:30:28           {
2023-08-16 23:30:28             "$k": [
2023-08-16 23:30:28               "$match"
2023-08-16 23:30:28             ],
2023-08-16 23:30:28             "$match": {
2023-08-16 23:30:28               "$k": [
2023-08-16 23:30:28                 "GroupType"
2023-08-16 23:30:28               ],
2023-08-16 23:30:28               "GroupType": {
2023-08-16 23:30:28                 "$k": [
2023-08-16 23:30:28                   "$in"
2023-08-16 23:30:28                 ],
2023-08-16 23:30:28                 "$in": [
2023-08-16 23:30:28                   0
2023-08-16 23:30:28                 ]
2023-08-16 23:30:28               }
2023-08-16 23:30:28             }
2023-08-16 23:30:28           },
2023-08-16 23:30:28           {
2023-08-16 23:30:28             "$k": [
2023-08-16 23:30:28               "$project"
2023-08-16 23:30:28             ],
2023-08-16 23:30:28             "$project": {
2023-08-16 23:30:28               "$k": [
2023-08-16 23:30:28                 "Id",
2023-08-16 23:30:28                 "Name",
2023-08-16 23:30:28                 "Group_Id",
2023-08-16 23:30:28                 "GroupType",
2023-08-16 23:30:28                 "_id"
2023-08-16 23:30:28               ],
2023-08-16 23:30:28               "Id": "$_id",
2023-08-16 23:30:28               "Name": "$Name",
2023-08-16 23:30:28               "Group_Id": "$Group_Id",
2023-08-16 23:30:28               "GroupType": "$GroupType",
2023-08-16 23:30:28               "_id": 0
2023-08-16 23:30:28             }
2023-08-16 23:30:28           }
2023-08-16 23:30:28         ],
2023-08-16 23:30:28         "cursor": {
2023-08-16 23:30:28           "$k": []
2023-08-16 23:30:28         },
2023-08-16 23:30:28         "$db": "ApplicationConfig"
2023-08-16 23:30:28       },
2023-08-16 23:30:28       "Kind": 0
2023-08-16 23:30:28     }
2023-08-16 23:30:28   ]
2023-08-16 23:30:28 }
2023-08-16 23:30:28 
2023-08-16 23:30:28 
2023-08-16 23:30:28 
2023-08-16 23:30:28 2023-08-16T21:30:28.117Z    DEBUG   pg/pg.go:119    DBPool: found existing pool     {"username": ""}
2023-08-16 23:30:28 2023-08-16T21:30:28.118Z    WARN    // 172.17.0.1:39752 -> 172.17.0.2:27017         clientconn/conn.go:598  Response header: length:   145, id:   13, response_to:   41, opcode: OP_MSG
2023-08-16 23:30:28 github.com/FerretDB/FerretDB/internal/clientconn.(*conn).logResponse
2023-08-16 23:30:28     /src/internal/clientconn/conn.go:598
2023-08-16 23:30:28 github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run
2023-08-16 23:30:28     /src/internal/clientconn/conn.go:311
2023-08-16 23:30:28 github.com/FerretDB/FerretDB/internal/clientconn.acceptLoop.func1
2023-08-16 23:30:28     /src/internal/clientconn/listener.go:321
2023-08-16 23:30:28 2023-08-16T21:30:28.118Z    WARN    // 172.17.0.1:39752 -> 172.17.0.2:27017         clientconn/conn.go:599  Response message:
2023-08-16 23:30:28 {
2023-08-16 23:30:28   "Checksum": 0,
2023-08-16 23:30:28   "FlagBits": 0,
2023-08-16 23:30:28   "Sections": [
2023-08-16 23:30:28     {
2023-08-16 23:30:28       "Document": {
2023-08-16 23:30:28         "$k": [
2023-08-16 23:30:28           "ok",
2023-08-16 23:30:28           "errmsg",
2023-08-16 23:30:28           "code",
2023-08-16 23:30:28           "codeName"
2023-08-16 23:30:28         ],
2023-08-16 23:30:28         "ok": {
2023-08-16 23:30:28           "$f": 0
2023-08-16 23:30:28         },
2023-08-16 23:30:28         "errmsg": "Cannot do exclusion on field _id in inclusion projection",
2023-08-16 23:30:28         "code": 31254,
2023-08-16 23:30:28         "codeName": "Location31254"
2023-08-16 23:30:28       },
2023-08-16 23:30:28       "Kind": 0
2023-08-16 23:30:28     }
2023-08-16 23:30:28   ]
2023-08-16 23:30:28 }
2023-08-16 23:30:28 
2023-08-16 23:30:28 
2023-08-16 23:30:28 
2023-08-16 23:30:28 github.com/FerretDB/FerretDB/internal/clientconn.(*conn).logResponse
2023-08-16 23:30:28     /src/internal/clientconn/conn.go:599
2023-08-16 23:30:28 github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run
2023-08-16 23:30:28     /src/internal/clientconn/conn.go:311
2023-08-16 23:30:28 github.com/FerretDB/FerretDB/internal/clientconn.acceptLoop.func1
2023-08-16 23:30:28     /src/internal/clientconn/listener.go:321
````

**Comment by BoBiene** [comment]
Looks like this one is fixed now with the release of the [1.10](https://github.com/FerretDB/FerretDB/releases/tag/v1.10.1)

**Comment by princejha95** [comment]
Will be picking up this @AlekSi
